nifi-registry/nifi-registry-core/nifi-registry-web-api/src/main/java/org/apache/nifi/registry/web/security/authentication/jwt/JwtService.java

```diff
@@ -82,6 +85,27 @@ public String getUserIdentityFromToken(final String base64EncodedToken) throws J
         }
     }
 
+    public Set<String> getUserGroupsFromToken(final String base64EncodedToken) throws JwtException {
```

Comment:
As mentioned on the calling reference, decoding, parsing, and verifying the token more than once is a concern because it happens on every HTTP request. Evaluating an alternative approach that parses and verifies the token, returning a structured set of claims, would be a better path forward.

Reply:
Refactored to improve the efficiency of token parsing / validation. I did change the method signature of a couple public methods - if you'd rather me not do that just let me know and I'll just add new methods instead of replacing the existing ones.